### PR TITLE
Fixed unreported strict property initialization violations.

### LINKED ISSUE #35884: Definite assignment checks don't occur on private fields
https://www.typescriptlang.org/play/index.html?ts=3.8.0-pr-30829-97#code/MYGwhgzhAEDC0G8BQ1oGIBmB7LAuaEALgE4CWAdgOYDcSK0ARuABYAUAlIvaoc6RADpMOWqgC+SMUA

```ts
// @strictPropertyInitialization: true
// @strictNullChecks: true

class C {
  #foo: string;

  blah() {
    this.#foo;
  }
}
```

**Expected**

```
Property 'foo' has no initializer and is not definitely assigned in the constructor.
```

**Actual**

No error

### PR DESCRIPTION
**Pull request by dragomirtitian**
Fixes #35884

### CODE REVIEW

File: tests/cases/conformance/classes/members/privateNames/privateNamesUnique-1.ts [hunk 1]
```diff
@@ -1,5 +1,6 @@
 // @strict: true
 // @target: es6
+// @strictPropertyInitialization: false
```
Comment (mheiber) [inline tests/cases/conformance/classes/members/privateNames/privateNamesUnique-1.ts]: I like how we're testing this code path as well 👍

Comment (dragomirtitian) [reply tests/cases/conformance/classes/members/privateNames/privateNamesUnique-1.ts]: 😅 I have to confess it was more not wanting to change existing tests too much, but it works out I think 😊. The main tests for the option seem good enough and cluttering other tests with extra errors seemed pointless.

**Comment by dragomirtitian** [comment]
@DanielRosenwasser Any chance to get this merged before the beta goes out ?

**Comment by RyanCavanaugh** [comment]
@dragomirtitian many thanks!

**Pull request merged**